sql/testdata/txn

```diff
@@ -92,6 +92,21 @@ SELECT * FROM kv
 ----
 a c
 
+# Abort transaction with a syntax error, and ignore statements until the end of the transaction block
+
+statement ok
+BEGIN
+
+query error syntax error at or near ","
+SELECT COUNT(*, 1) FROM kv
+----
+
+statement error current transaction is aborted, commands ignored until end of transaction block
+UPDATE kv SET v = 'b' WHERE k in ('a')
+
+statement ok
+ROLLBACK
+
```

Comment:
FYI mysql doesn't abort transactions after a syntax error; it just returns the error and lets the application/user decide whether to continue the transaction or abort. This is really handy at the interactive prompt since you don't have to restart if you make a typo, although I'm not sure how well we'll be able support this style of usage given our reliance on transaction restarts in general.

```
mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> select count(*,1) from test;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '1) from test' at line 1
mysql> insert into test values (1);
Query OK, 1 row affected (0.00 sec)

mysql> commit;
Query OK, 0 rows affected (0.00 sec)
```